samples/client/petstore/java/native/src/main/java/org/openapitools/client/api/AnotherFakeApi.java

```diff
@@ -106,10 +106,11 @@ public ApiResponse<Client> call123testSpecialTagsWithHttpInfo(Client body) throw
         if (localVarResponse.statusCode()/ 100 != 2) {
           throw getApiException("call123testSpecialTags", localVarResponse);
         }
+        InputStream responseBody = localVarResponse.body();
         return new ApiResponse<Client>(
           localVarResponse.statusCode(),
           localVarResponse.headers().map(),
-          memberVarObjectMapper.readValue(localVarResponse.body(), new TypeReference<Client>() {}) // closes the InputStream
+          responseBody == null || responseBody.available() < 1 ? null : memberVarObjectMapper.readValue(responseBody, new TypeReference<Client>() {}) // closes the InputStream
```

Comment:
Seems like [this implementation](https://github.com/openjdk/jdk11u/blob/master/src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java#L304) of `InputStream` would be used here. This implementation doesn't seems to support `mark` & `reset` methods, therefore only variant that I found is to use `available` here. Happy to hear any advice to how this check can be implemented better

Reply:
thanks for pointing it out.

my take is to get it merged and let the users to try it out to see if there are better ways.